code/src/components/app-input-with-icon/AppInputWithIcon.types.ts

```diff
@@ -0,0 +1,7 @@
+import { InputBaseProps } from "@mui/material";
+
+
+export type InputWithIconProps = InputBaseProps & {
+    onSearch?: () => void;
+    onClear?: () => void;
+  };
```

Comment:
I can see that this file lacks formatting. If I am mistaken, please correct me

Reply:
Formated!